### PR TITLE
[Logs Bridge API] Add support for Body set directly on LogRecord for OTLP Exporter.

### PR DESCRIPTION
**Pull request by Yun-Ting**
## Changes

Towards https://github.com/open-telemetry/opentelemetry-dotnet/issues/4433
Added support for `Body` set directly on `LogRecord` via the Logs Bridge API.
## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs [hunk 1]
```diff
@@ -164,12 +165,21 @@ internal OtlpLogs.LogRecord ToOtlpLog(LogRecord logRecord)
...
+                        useTemplateFromExtensionMethod = true;
                     }
                     else if (OtlpKeyValueTransformer.Instance.TryTransformTag(attribute, out var result, attributeValueLengthLimit))
                     {
                         AddAttribute(otlpLogRecord, result, attributeCountLimit);
                     }
                 }
+
+                // Supports Body set directly on LogRecord for the Logs Bridge API.
+                if (!useTemplateFromExtensionMethod && !bodyPopulatedFromFormattedMessage && logRecord.Body != null)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:177]: I would probably rename `useTemplateFromExtensionMethod` -> `bodyPopulatedFromMessageTemplate` but why not just...

```csharp
if (otlpLogRecord.Body == null && logRecord.Body != null)
{
   otlpLogRecord.Body = new OtlpCommon.AnyValue { StringValue = logRecord.Body };
}
```

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs [hunk 2]
```diff
@@ -589,6 +589,34 @@ public void CheckToOtlpLogRecordBodyIsPopulated(bool includeFormattedMessage)
         Assert.Equal("state", otlpLogRecord.Body.StringValue);
     }
 
+#if EXPOSE_EXPERIMENTAL_FEATURES && NET8_0_OR_GREATER
```
Comment (CodeBlanch) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs:592]: I don't think `EXPOSE_EXPERIMENTAL_FEATURES` works in test projects. Typically they use `internal`s when things aren't exposed. But it looks like we don't do `InternalsVisibleTo` from in-memory exporter. Here's a way you can make it work which shouldn't need any preprocessor directives or extra `InternalsVisibleTo` definitions:

```csharp
    [Fact]
    public void LogRecordBodyIsExportedWhenUsingBridgeApi()
    {
        var logRecords = new List<LogRecord>();

        using (var loggerProvider = Sdk.CreateLoggerProviderBuilder()
            .ConfigureServices(services =>
            {
                services.AddLogging(builder => builder
                    .AddOpenTelemetry(options => options.AddInMemoryExporter(logRecords)));
            })
            .Build())
        {
            var logger = loggerProvider.GetLogger();

            logger.EmitLog(new LogRecordData()
            {
                Body = "Hello world",
            });
        }

        Assert.Single(logRecords);

        var otlpLogRecordTransformer = new OtlpLogRecordTransformer(DefaultSdkLimitOptions, new());

        var otlpLogRecord = otlpLogRecordTransformer.ToOtlpLog(logRecords[0]);

        Assert.Equal("Hello world", otlpLogRecord.Body?.StringValue);
    }
```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs [hunk 3]
```diff
@@ -170,6 +170,15 @@ internal OtlpLogs.LogRecord ToOtlpLog(LogRecord logRecord)
                         AddAttribute(otlpLogRecord, result, attributeCountLimit);
                     }
                 }
+#if EXPOSE_EXPERIMENTAL_FEATURES && NET8_0_OR_GREATER
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:173]: You shouldn't need these preprocessor directives here.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* **Experimental (pre-release builds only):** Added
+  support for `Body` set directly on `LogRecord` via the Logs Bridge API.
+  [#5268](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5268)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:7]: nit: Add this entry below the other things already in the unreleased section

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 5]
```diff
@@ -15,6 +15,10 @@
   will be automatically included in exports.
   ([#5258](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5258))
 
+* **Experimental (pre-release builds only):** Added
+  support for `Body` set directly on `LogRecord` via the Logs Bridge API.
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:19]:
```suggestion
 support for setting `Body` directly on `LogRecord` via the Logs Bridge API.
```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs [hunk 6]
```diff
@@ -170,6 +170,14 @@ internal OtlpLogs.LogRecord ToOtlpLog(LogRecord logRecord)
                         AddAttribute(otlpLogRecord, result, attributeCountLimit);
                     }
                 }
+
+                // Supports Body set directly on LogRecord for the Logs Bridge API.
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:174]:
```suggestion
                // Supports setting Body directly on LogRecord for the Logs Bridge API.
```

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs [hunk 7]
```diff
@@ -589,6 +589,38 @@ public void CheckToOtlpLogRecordBodyIsPopulated(bool includeFormattedMessage)
...
 
+    [Fact]
+    public void LogRecordBodyIsExportedWhenUsingBridgeApi()
+    {
+        var logRecords = new List<LogRecord>();
+
+        using (var loggerProvider = Sdk.CreateLoggerProviderBuilder()
+            .ConfigureServices(services =>
+            {
+                services.AddLogging(builder => builder.AddOpenTelemetry(options =>
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs:600]: curious why this pattern? why not

```csharp
Sdk.CreateLoggerProviderBuilder()
.AddInMemoryExporter(logRecords)
.Build();
```

Comment (vishweshbankwar) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs:600]: Missed this https://github.com/open-telemetry/opentelemetry-dotnet/pull/5268#discussion_r1468249286. @Yun-Ting Could you please add a comment there on why are using this pattern. 
I think we could consider internalsvisibleto from InMemory Exporter - @CodeBlanch what do you think?

**Review by vishweshbankwar — APPROVED** [review]
LGTM: left couple nits.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs (hunk 7 above)
Comment (CodeBlanch) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs]: Works for me! Updated

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs [hunk 8]
```diff
@@ -589,6 +589,32 @@ public void CheckToOtlpLogRecordBodyIsPopulated(bool includeFormattedMessage)
         Assert.Equal("state", otlpLogRecord.Body.StringValue);
     }
 
+    [Fact]
+    public void LogRecordBodyIsExportedWhenUsingBridgeApi()
```
Comment (rajkumar-rangaraj) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs:593]: Should we add a test case to prove what happens when `otlpLogRecord.Body` is already set?

Comment (Yun-Ting) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs]: added!

**Pull request merged**